### PR TITLE
[dev-task] standardize & fix fromVersion validations and formatting

### PR DESCRIPTION
**Pull request by MLainer1**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-572)

## Description
Remove support for the fromServerVersion entry, and replace it with the fromVersion entry

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

**Review by idovandijk — COMMENTED** [review]
@MLainer1 what is the implication of no longer supporting this? If I'm not mistaken `format` already removes that field and validation passes, am I wrong?
I just want to make sure that if we are gonna throw errors with `validate` we would also handle the case with `format`

**Comment by dorschw** [comment]
@idovandijk please see the related issue, and the ones linked from there.
Format does not yet remove `fromServerVersion` & co., this will be part of this PR. 
Any failing validation will come with a matching fix in `format`.

**Comment by MLainer1** [comment]
> @MLainer1 what is the implication of no longer supporting this? If I'm not mistaken `format` already removes that field and validation passes, am I wrong? I just want to make sure that if we are gonna throw errors with `validate` we would also handle the case with `format`

We still have some content items that have the fromServerVersion entry, so this is a way to handle that scenario in the `format` command, and make use of the fromServerValue and not just remove it

File: demisto_sdk/commands/common/hook_validations/lists.py [hunk 1]
```diff
@@ -43,15 +43,15 @@ def is_valid_version(self) -> bool:
         return self._is_valid_version()
 
     @error_codes('LI100,LI101')
-    def is_valid_from_server_version(self) -> bool:
+    def is_valid_server_version(self) -> bool:
```
Comment (dantavori) [inline demisto_sdk/commands/common/hook_validations/lists.py:46]: why did we remove the "from" here?

File: demisto_sdk/commands/format/update_generic.py [hunk 2]
```diff
@@ -347,3 +356,28 @@ def sync_data_to_master(self):
...
+        if old_from_server_version and not current_from_server_version and not current_from_version:
+            self.data[self.from_version_key] = old_from_server_version
+        elif current_from_server_version and not current_from_version:
+            self.data[self.from_version_key] = current_from_server_version
+            self.data.pop(self.from_server_version_key)
+        elif current_from_server_version and current_from_version:
+            if current_from_server_version == current_from_version:
+                self.data.pop(self.from_server_version_key)
+            else:
+                preserve_from_version = click.confirm(f'Both "{self.from_version_key}" and "{self.from_server_version_key}" '
```
Comment (dantavori) [inline demisto_sdk/commands/format/update_generic.py:376]: see `set_default_from_version()` for user input handling/assume yes

File: demisto_sdk/commands/format/update_generic.py [hunk 3]
```diff
@@ -347,3 +356,28 @@ def sync_data_to_master(self):
...
+        old_from_server_version = self.old_file.get(self.from_server_version_key)
+
+        if old_from_server_version and not current_from_server_version and not current_from_version:
+            self.data[self.from_version_key] = old_from_server_version
+        elif current_from_server_version and not current_from_version:
+            self.data[self.from_version_key] = current_from_server_version
+            self.data.pop(self.from_server_version_key)
+        elif current_from_server_version and current_from_version:
+            if current_from_server_version == current_from_version:
+                self.data.pop(self.from_server_version_key)
```
Comment (dantavori) [inline demisto_sdk/commands/format/update_generic.py:374]: nice

File: demisto_sdk/commands/format/update_generic.py [hunk 4]
```diff
@@ -295,6 +296,14 @@ def set_from_version_key_name(self) -> Union[str, None]:
             return 'fromVersion'
         return None
 
+    def set_from_server_version_key_name(self) -> Union[str, None]:
+        """fromversion key is different between yml and json , in yml file : fromserverversion, in json files : fromServerVersion"""
+        if self.file_type == "yml":
+            return 'fromserverversion'
```
Comment (dantavori) [inline demisto_sdk/commands/format/update_generic.py:302]: did we check if needed?

Comment (MLainer1) [reply demisto_sdk/commands/format/update_generic.py:302]: Not needed, will remove

File: demisto_sdk/commands/format/update_generic.py [hunk 5]
```diff
@@ -75,6 +75,7 @@ def __init__(self,
         except Exception:
             raise Exception(F'Provided file {self.source_file} is not a valid file.')
         self.from_version_key = self.set_from_version_key_name()
+        self.json_from_server_version_key = "fromServerVersion"
```
Comment (dantavori) [inline demisto_sdk/commands/format/update_generic.py:78]: let's move this to constants.py

**Review by dantavori — APPROVED** [review]

**Pull request merged**